common/commands.cpp

```diff
@@ -0,0 +1,122 @@
+#include "commands.h"
+
+namespace fty::disco::commands::scan {
+
+namespace start {
+    std::ostream& operator<<(std::ostream& ss, In::Type value)
+    {
+        using Type = In::Type;
+
+        ss << [&]() {
+            switch (value) {
+                case Type::Local:
+                    return "localscan";
+                case Type::Ip:
+                    return "ipscan";
+                case Type::Multy:
+                    return "multiscan";
+                case Type::Full:
+                    return "fullscan";
+                case Type::Unknown:
+                    return "UNKNOWN";
+            }
+            return "Unknown";
+        }();
+        return ss;
+    }
+
+    std::istream& operator>>(std::istream& ss, In::Type& value)
+    {
+        using Type = In::Type;
+
+        std::string strval;
+        ss >> strval;
+        if (strval == "localscan") {
+            value = Type::Local;
+        } else if (strval == "ipscan") {
+            value = Type::Ip;
+        } else if (strval == "multiscan") {
+            value = Type::Multy;
+        } else if (strval == "fullscan") {
+            value = Type::Full;
+        } else {
+            value = Type::Unknown;
+        }
+        return ss;
+    }
+} // namespace start
+
+namespace status {
+    std::ostream& operator<<(std::ostream& ss, Out::Status value)
+    {
+        using Status = Out::Status;
+
+        ss << [&]() {
+            switch (value) {
+                case Status::CancelledByUser:
+                    return "1";
+                case Status::Terminated:
+                    return "2";
+                case Status::InProgress:
+                    return "3";
+                case Status::Unknown:
+                    return "0";
+            }
```

Comment:
at least 
```
ss << fty::convert<int>(value);
```
:)
But I don't understand why in this special case you need numeric values :(